core/templates/pages/exploration-editor-page/improvements-tab/services/improvement-modal.service.ts

```diff
@@ -59,165 +56,15 @@ angular.module('oppia').factory('ImprovementModalService', [
-              };
-
-              /**
-               * Renders a table for MultipleIncorrectSubmissions issue for a
-               * more friendly UI.
-               */
-              $scope.renderIssueTable = function() {
-                var lars = LearnerActionRenderService;
-                var tableHtml =
-                  lars.renderFinalDisplayBlockForMISIssueHTML(
-                    playthrough.actions.slice(
-                      actionIndexAtStartOfFinalBlock),
-                    actionIndexAtStartOfFinalBlock + 1);
-                return tableHtml;
-              };
-
-              /**
-               * Returns the list of learner actions to currently be displayed
-               * in the playthrough modal. If the issue is a Multiple Incorrect
-               * Submissions one, we return the list of learner actions,
-               * excluding only the final display block (since we are rendering
-               * a table for the multiple incorrect submissions instead).
-               * @returns {LearnerAction[]}
-               */
-              $scope.getActionsToRender = function() {
-                if ($scope.issueIsMultipleIncorrectSubmissions) {
-                  return playthrough.actions.slice(
-                    indexOfFirstDisplayedAction,
-                    actionIndexAtStartOfFinalBlock);
-                }
-                return playthrough.actions.slice(indexOfFirstDisplayedAction);
-              };
-
-              /**
-               * Returns the index of the learner action.
-               * @param {int} displayedActionIndex.
-               * @returns {int}
-               */
-              $scope.getLearnerActionIndex = function(displayedActionIndex) {
-                return indexOfFirstDisplayedAction + displayedActionIndex + 1;
-              };
-
-              /**
-               * Computes whether the learner action needs to be highlighted.
-               * @param {int} actionIndex.
-               * @returns {boolean}
-               */
-              $scope.isActionHighlighted = function(actionIndex) {
-                return $scope.getLearnerActionIndex(actionIndex) > (
-                  actionIndexAtStartOfFinalBlock);
-              };
-
-              /**
-               * Renders the HTML of the learner action.
-               * @param {LearnerAction} learnerAction.
-               * @param {int} actionIndex - The index of the learner action.
-               * @returns {string}
-               */
-              $scope.renderLearnerAction = function(
-                  learnerAction, actionIndex) {
-                return LearnerActionRenderService.renderLearnerAction(
-                  learnerAction, actionIndex);
-              };
-
-              $scope.cancel = function() {
-                $uibModalInstance.dismiss('cancel');
-                AlertsService.clearWarnings();
-              };
-
-              init();
+          resolve: {
+            playthrough: function() {
```

Comment:
Please use fat arrow functions for these tiny return-only logic everywhere in this PR.

Reply:
Done.